### ISSUE TITLE
If transferred to a submarine with a shuttle, then you will permanently lose your lovely shuttle :(

### ISSUE THREAD

**Issue by esirprus** [Regalis11/Barotrauma#9380]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![image](https://user-images.githubusercontent.com/44423036/174212413-6d1b0460-0134-4c92-bf99-105f02ec279a.png)
![image](https://user-images.githubusercontent.com/44423036/174212443-da581c20-b166-4c66-9821-43ae343852f6.png)
![image](https://user-images.githubusercontent.com/44423036/174212518-6fe116f3-510b-4fb9-b0d5-234d2ef9d67f.png)

### Reproduction steps

1. start a campaign, and choose a vanilla sub.
2. purchase and swith to kastrull (transfer items)
3. leave the outpost
4. try to find you shuttle, you can find your shuttle in the wall :(

### Bug prevalence

Happens every time I play

### Version

0.18.11.0

### Which operating system did you encounter this bug on?

Windows

**Comment by esirprus** [comment]
@Regalis11 plz see this issue, plz

**Comment by Regalis11** [comment]
Thank you for the report! Fixed in https://github.com/Regalis11/Barotrauma-development/commit/decf9ed6f8dac8ba017e2c980070e6d6aebd7ae8

**Comment by NilanthAnimosus** [comment]
Tested on 0.18.13.0, works correctly on new saves only.

It appears new campaigns are just fine now and no longer lose their linked shuttle to loading issues.

However, I Went back to 0.18.11.0 and made a campaign there, triggered the bug and brought that save to the current version getting these errors (Which I also got in that version):
![image](https://user-images.githubusercontent.com/29177976/176851048-f50365df-624f-4668-80b5-5a6cb87dd012.png)

Note: I started the round then went to main menu, so the point that save file is at is when leaving the outpost. I assume the save file after that point may be permanently corrupted and not have anything saved for the shuttle as I cannot actually visually see it (even in a wall anywhere) anymore.

Using this single player save file: [Shuttle Test.zip](https://github.com/Regalis11/Barotrauma/files/9027148/Shuttle.Test.zip)
